### PR TITLE
[GPU] Remove use of legacy names

### PR DESCRIPTION
**Pull request by Lyamin-Roman**
### Tickets:
 - *132084*

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/plugin/program_builder.hpp [hunk 1]
```diff
@@ -143,6 +149,7 @@ class ProgramBuilder final {
 private:
     static factories_map_t factories_map;
     std::shared_ptr<cldnn::program> m_program;
+    std::shared_ptr<ov::Model> m_model;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/include/intel_gpu/plugin/program_builder.hpp:152]: does it  mean that `ov::Model` will live forever after being compiled to `ov::CompiledModel` ?

File: src/plugins/intel_gpu/src/plugin/ops/result.cpp [hunk 2]
```diff
@@ -20,12 +20,9 @@ static void CreateResultOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v0::
...
     auto prev = op->get_input_node_shared_ptr(0);
-    auto input_id = ov::op::util::get_ie_output_name(op->get_input_source_output(0));
-    if (input_id.empty()) {
-        input_id = prev->get_friendly_name();
-        if (prev->get_output_size() > 1) {
-            input_id += "." + std::to_string(op->get_input_source_output(0).get_index());
-        }
+    auto input_id = prev->get_friendly_name();
+    if (prev->get_output_size() > 1) {
+        input_id += "." + std::to_string(op->get_input_source_output(0).get_index());
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/ops/result.cpp:25]: why do we need this legacy logic? Result has its own name, which can be used here (for some reason)

File: src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp [hunk 3]
```diff
@@ -209,24 +195,21 @@ void SyncInferRequest::set_tensors_impl(const ov::Output<const ov::Node> port, c
...
     OPENVINO_ASSERT(is_host || is_remote, "[GPU] Incorrect input blobs. All blobs must be of the same type");
 
-    for (const auto& input : get_inputs()) {
-        if (input == port) {
-            m_batched_tensors[input.get_tensor_ptr()] = tensors;
-            return;
-        }
-    }
-    OPENVINO_THROW("[GPU] Cannot find input tensors for port ", port);
+    size_t port_index = find_port(port).idx;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp:198]: find_port return type of port, which we can also check here

File: src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp [hunk 4]
```diff
@@ -209,24 +195,21 @@ void SyncInferRequest::set_tensors_impl(const ov::Output<const ov::Node> port, c
...
+    OPENVINO_ASSERT(m_input_ports_map.count(port_index) != 0, "[GPU] Cannot find input tensors for port ", port, " with index ", port_index);
+    const auto& tensor = m_input_ports_map.at(port_index).get_tensor_ptr();
+    m_batched_tensors[tensor] = tensors;
 }
 
 ov::SoPtr<ov::ITensor> SyncInferRequest::get_tensor(const ov::Output<const ov::Node>& port) const {
     bool is_input = ov::op::util::is_parameter(port.get_node());
-    const auto name = get_port_name(port);
+    size_t port_index = find_port(port).idx;
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp:206]: let's drop usage of `ov::op::util::is_parameter` and rely on port_type returned by `find_port`

File: src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp [hunk 5]
```diff
@@ -265,23 +248,24 @@ void SyncInferRequest::enqueue() {
     std::vector<cldnn::event::ptr> dependencies;
 
     for (const auto& it : m_input_ports_map) {
-        const auto& name = it.first;
+        size_t port_idx = it.first;
         const auto& port = it.second;
 
         if (m_batched_tensors.count(port.get_tensor_ptr()) > 0) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/sync_infer_request.cpp:254]: is `port.get_tensor_ptr()` safe to use? should we switch to indexes as well?

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**